### DISCUSSION TITLE
Version 2.2.23 on ESP8266-01S

### DISCUSSION THREAD

**Discussion by gellfuen** [UtilitechAS/amsreader-firmware#661] [Q&A] [answered]
Hello community!

Try to flash amsreader firmware.bin (version ams2mqtt-esp8266-2.2.23) on a ESP8266-01S with esptool.py running on windows.

Flashing seems to work:

_python.exe esptool.py --chip esp8266 --port COM4 --baud 115200 --before default_reset --after hard_reset write_flash 0x0 firmware.bin
esptool.py v4.6.2
Serial port COM4
Connecting....
Chip is ESP8266EX
Features: WiFi
Crystal is 26MHz
MAC: 08:3a:8d:e7:e7:c5
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Flash will be erased from 0x00000000 to 0x000d7fff...
Compressed 884528 bytes to 588789...
Wrote 884528 bytes (588789 compressed) at 0x00000000 in 53.2 seconds (effective 133.1 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin..._

But after disconnecting from the USB adapter and reconnecting WLAN is NOT coming up and I only can see this at terminal:

 ets Jan  8 2013,rst cause:4, boot mode:(3,2)

wdt reset
ets_main.c 

Any idea what is going on?
Is that tiny ESP8266-01S supported?

The hardware works with different firmware file...
 
Best regards

**Comment by ArnieO** [comment]
It's been a really longe time since I worked with ESP-01, but...

Are you sure that ESP-01s has enough flash memory - is it 1 MB? 
The firmware is designed for ESP12e/f, which has 4 MB. 

Unless you use an ESP8266 module with 4 MB flash memory, you cannot expect it to work.

**Comment by gellfuen** [comment]
Hello ArnieO,

Oh, I did not know, that it was designed for 4MB only! I had that part lying around, so I used it.

THX for this answer :-).

Decryption of the data from the smart meter should be possible with an ESP8266 anyway?

In the meantime I flashed a brand new D1 mini with 4MB and a ESP8266-12F on it.
Much more is coming back from that in terminal, but I cannot see the WLAN AP of it, and in terminal I saw the message WLAN not connected!
Any hint on this?
best regards

**Reply by ArnieO** [reply, marked as answer]
Yes, it works on ESP8266 - but your device need to have enough flash memory. 

For information on how to use the firmware, I advise you to read the Firmware User Manual that can be downloaded from the product pages on https://amsleser.no.
